worksheet_D/oxo.py

```diff
@@ -0,0 +1,107 @@
+class OxoBoard:
+    def __init__(self):
+        # Initialising board array
+        self.boardList = [[0, 0, 0], [0, 0, 0], [0, 0, 0]]
+
+    def get_square(self, x, y):
+        # Returns value of board square
+        return self.boardList[y][x]
+
+    def set_square(self, x, y, mark):
+        # Fills square with mark, unless it's already filled
+        if self.boardList[y][x] == 0:
+            self.boardList[y][x] = mark
+            return True
+        else:
+            return False
+
+    def is_board_full(self):
+        # Returns false if any of the squares contain 0
+        for x in xrange(3):
+            for y in xrange(3):
+                if self.boardList[y][x] == 0:
+                    return False
+        return True
+
+    def get_winner(self):
+        # Checks rows, columns and diagonals for 3 matching squares that aren't 0 and returns the square value
+        for i in xrange(3):
+            # Horizontal matches
+            if self.boardList[i][0] == self.boardList[i][1] == self.boardList[i][2] and self.boardList[i][0] > 0:
+                return self.boardList[i][0]
+            # Vertical matches
+            elif self.boardList[0][i] == self.boardList[1][i] == self.boardList[2][i] and self.boardList[0][i] > 0:
+                return self.boardList[0][i]
+            # Diagonal matches
+            elif self.boardList[0][0] == self.boardList[1][1] == self.boardList[2][2] and self.boardList[0][0] > 0:
```

Comment:
Diagonal checks don't need to be inside the loop -- you're checking the same thing 3 times when once would do